### PR TITLE
changes to make pharo site work with foliage

### PR DESCRIPTION
**Pull request by estebanlm**
- blogs can have a title and a layout
- blog posts produce an abstract using the same mechanism than the rest (we could change that abstract)
- FOCSSObject is now a more generic FOFileObject
- implement some missings in html visitor
- recategorisations

### CODE REVIEW

**Comment by estebanlm** [comment]
this version requires "Macrodown", an extended version of Microdown, btw :)

File: src/Foliage-Core/FOBlogOverviewBuilder.class.st [hunk 1]
```diff
@@ -2,6 +2,7 @@ Class {
 	#name : #FOBlogOverviewBuilder,
 	#superclass : #Object,
 	#instVars : [
+		'blog',
 		'posts',
```
Comment (noha) [inline src/Foliage-Core/FOBlogOverviewBuilder.class.st]: isn't posts superfluous if blog is there?

File: src/Foliage-Core/FOBlogPage.class.st [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Class {
+	#name : #FOBlogPage,
```
Comment (noha) [inline src/Foliage-Core/FOBlogPage.class.st:2]: What is the intention of a blog page? The list of abstracts?

File: src/Foliage-Core/FOBlogPost.class.st [hunk 3]
```diff
@@ -4,17 +4,66 @@ Class {
...
 
-{ #category : #'as yet unclassified' }
+{ #category : #accessing }
+FOBlogPost >> abstract [
+
+	^ super abstract
+]
+
+{ #category : #accessing }
+FOBlogPost >> abstractLayout [
```
Comment (noha) [inline src/Foliage-Core/FOBlogPost.class.st:14]: I think I need to work here a bit. The structure of the document should usable and every part can have its own layout

File: src/Foliage-Core/FOBlogPost.class.st [hunk 4]
```diff
@@ -4,17 +4,66 @@ Class {
...
+
+{ #category : #visiting }
 FOBlogPost >> acceptFOVisitor: aFOVisitor [ 
 	^ aFOVisitor visitBlogPost: self 
 ]
 
+{ #category : #accessing }
+FOBlogPost >> defaultAbstractLayout [
+
+	^ 'blogpostabstract'
```
Comment (noha) [inline src/Foliage-Core/FOBlogPost.class.st:35]: meh! ;)

File: src/Foliage-Core/FOBlogPostRenderContext.class.st [hunk 5]
```diff
@@ -4,12 +4,36 @@ Class {
...
+	^ page pathString
+]
+
 { #category : #'as yet unclassified' }
 FOBlogPostRenderContext >> previousPost [ 
 	^ page parent postBefore: self page
 ]
+
+{ #category : #accessing }
+FOBlogPostRenderContext >> publicationDate [
```
Comment (noha) [inline src/Foliage-Core/FOBlogPostRenderContext.class.st]: This should be available for all pages/published items. This is the switch to make it public

File: src/Foliage-Server/FOServer.class.st [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+FOServer class >> stop [
+	<script>
+	
+	^ ZnServer default stop
+]
+
+{ #category : #accessing }
+FOServer >> baseDirectory [
+
+	^ 'html' asFileReference
```
Comment (noha) [inline src/Foliage-Server/FOServer.class.st]: we is that "html" when the target path above is "generated"?

**Review by noha — APPROVED** [review]

File: src/Foliage-Core/FOBlogOverviewBuilder.class.st (hunk 1 above)
Comment (estebanlm) [reply src/Foliage-Core/FOBlogOverviewBuilder.class.st]: indeed

File: src/Foliage-Server/FOServer.class.st (hunk 6 above)
Comment (estebanlm) [reply src/Foliage-Server/FOServer.class.st]: oops

**Pull request merged**